<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>776</width>
    <height>394</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_5">
   <item>
    <widget class="OpenGLWidget" name="openGLWidget">
     <property name="minimumSize">
      <size>
       <width>369</width>
       <height>372</height>
      </size>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Values</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="toolTip">
             <string>Type of stairs to draw.</string>
            </property>
            <property name="text">
             <string>Stairs type:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox">
            <property name="toolTip">
             <string>Type of stairs to draw.</string>
            </property>
            <item>
             <property name="text">
              <string>Straight Stairs</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Platform Stairs</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>U Shaped Stairs</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="toolTip">
             <string>The width of every individual step.</string>
            </property>
            <property name="text">
             <string>Step width:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="stepWidth">
            <property name="toolTip">
             <string>The width of every individual step.</string>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>5.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="toolTip">
             <string>The height of every individual step.</string>
            </property>
            <property name="text">
             <string>Step height:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="stepHeight">
            <property name="toolTip">
             <string>The height of every individual step.</string>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="maximum">
             <double>5.000000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="toolTip">
             <string>The depth  of every individual step.</string>
            </property>
            <property name="text">
             <string>Step depth:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="stepDepth">
            <property name="toolTip">
             <string>The depth  of every individual step.</string>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="SegmentLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Number of steps for each segment</string>
            </property>
            <property name="text">
             <string>Stairs segment steps:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="segmentHeight">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Number of steps for each segment</string>
            </property>
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="toolTip">
             <string>The overall height of the stairs</string>
            </property>
            <property name="text">
             <string>Stairs height:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="stairsHeight">
            <property name="toolTip">
             <string>The overall height of the stairs</string>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>10.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="PlatformLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>The depth of the platforms</string>
            </property>
            <property name="text">
             <string>Platform depth:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="PlatformDepth">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>The depth of the platforms</string>
            </property>
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="value">
             <double>5.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <item>
           <widget class="QPushButton" name="resetCamera">
            <property name="toolTip">
             <string>Reset camera to starting position</string>
            </property>
            <property name="text">
             <string>Reset Camera</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Controls:</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Right Mouse Button: Rotate camera</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>W/A/S/D: Move camera(Keyboard language must be English)</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>openglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
